Introduce manager caching of Unstructured

erikgb opened the pull request:
Caching of Unstructured is disabled by default in controller-runtime, and that is suboptimal for HNC which is reconciling generic objects configured by users.

Tested: Ran both unit-tests ('make test') and integration test ('make test-e2e') successfully.

adrianludwin commented:
Also what do you think the risk of this is? Should we delay it to v1.1?

erikgb commented:
> Also what do you think the risk of this is? Should we delay it to v1.1?

Hmmm, hard to tell. 🤔 We have been caching Unstructured for a long time in our controllers, and there seems to be quite performance gain. But on the other hand, the controller memory consumption might increase. How much depends on the cluster since and how many "object" resource types are configured in HNC. But resources tuning should any cluster-admin be able to do....

I can start by pulling out the "fixes" into a separate PR that can be merged. And leave the Unstructured caching (this PR) open for now?

adrianludwin commented:
Yeah let's punt this to v1.1.

On Wed, Mar 16, 2022 at 6:55 PM Erik Godding Boye ***@***.***>
wrote:

> Also what do you think the risk of this is? Should we delay it to v1.1?
>
> Hmmm, hard to tell. 🤔 We have been caching Unstructured for a long time
> in our controllers, and there seems to be quite performance gain. But on
> the other hand, the controller memory consumption might increase. How much
> depends on the cluster since and how many "object" resource types are
> configured in HNC. But resources tuning should any cluster-admin be able to
> do....
>
> I can start by pulling out the "fixes" into a separate PR that can be
> merged. And leave the Unstructured caching (this PR) open for now?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes-sigs/hierarchical-namespaces/pull/153#issuecomment-1069721134>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AE43PZAG5NT5AHH7I6TXM5LVAJRERANCNFSM5QBODUHA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

rjbez17 commented:
/hold 

for 1.1

erikgb commented:
I have now rebased this and fixed commit message and PR description. So should be ready for review - even if we decide to wait for 1.1 with this.

/cc @rjbez17

adrianludwin reviewed (commented):
/lgtm
/approve

adrianludwin commented:
/hold cancel
